### PR TITLE
LRDOCS-4139 theme generator 7.1

### PR DESCRIPTION
**Pull request by mwilliams2014**
@Robert-Frampton These docs are the same as 7.0, just reorganized. I have plans for a third tutorial in this section that covers the Theme Generator's APIs. I'll need your help with that. For now though, for the initial release, I would like to update the theme tasks reference doc to include all the configuration options you have (flags, custom configuration parameters, etc.) Can you please provide me with a list of these?

### CODE REVIEW

File: develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        v8.9.4
+
+    +$$$
+
+    Note: To avoid any potential compatibility issues, we recommend installing
+    the Long Term Support (LTS) version of Node.js--at the time of this writing,
+    the LTS version is v8.9.4.
+    
+    $$$
+    <!-- Is this still recommended? -->
```
Comment (mwilliams2014) [inline develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown:30]: @Robert-Frampton is the LTS version still recommended?

File: develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    
+2.  @product@ themes require Compass to be used with Ruby based Sass. Install 
+    the Compass and Sass Ruby gems. Note that Sass version 3.5 has compatibility 
+    issues with Compass. We recommend that you install Sass version 3.4.0, as 
+    shown in the command below:
+    
+        gem install sass -v "=3.4.0"
+
+        gem install compass
+
```
Comment (mwilliams2014) [inline develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown:120]: @Robert-Frampton  Is Sass V 3.4.0 still the recommended version to install?

File: develop/reference/articles/05-theme-gulp-tasks.markdown [hunk 3]
```diff
@@ -54,6 +61,11 @@ Here are the gulp tasks you can execute:
     Mode](/develop/tutorials/-/knowledge_base/7-0/using-developer-mode-with-themes)
     enabled.
 
-**Related Topics**
+## Liferay Theme Generator Flag options
```
Comment (mwilliams2014) [inline develop/reference/articles/05-theme-gulp-tasks.markdown]: @Robert-Frampton what other flags/ configuration options are available for the theme generator? For instance, I know you can skip the update notifier and display deprecated options.

File: develop/reference/articles/05-theme-gulp-tasks.markdown [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
+such as the available gulp tasks for generated themes, optional flags for the 
+generator, etc.
 
-Here are the gulp tasks you can execute:
+## Theme Gulp Tasks [](id=theme-gulp-tasks)
 
-*  `build`: generates the base theme files, compiles Sass into CSS, and
-    zips all theme files into a WAR file that you can deploy to a Liferay
-    server. 
+Theme projects created using the
```
Comment (mwilliams2014) [inline develop/reference/articles/05-theme-gulp-tasks.markdown]: @Robert-Frampton I know the Server Manager Web plugin and --live deploy flag didn't work for 7.0. Will this be fixed for 7.1?

File: develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown (hunk 1 above)
Comment (robframpton) [reply develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown:30]: Yeah LTS is still recommended.

File: develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown (hunk 2 above)
Comment (robframpton) [reply develop/tutorials/articles/260-themes-and-layout-templates/01-themes/01-theme-generator/01-installing-the-generator.markdown:120]: I'm assuming that hasn't changed.

File: develop/reference/articles/05-theme-gulp-tasks.markdown (hunk 3 above)
Comment (robframpton) [reply develop/reference/articles/05-theme-gulp-tasks.markdown]: I don't think anything else has been added.

File: develop/reference/articles/05-theme-gulp-tasks.markdown (hunk 4 above)
Comment (robframpton) [reply develop/reference/articles/05-theme-gulp-tasks.markdown]: No, I don't that plugin will ever be used in future releases.

**Comment by robframpton** [comment]
Hey @izaera,

From a user perspective, has anything changed in the Themes SDK that's worth mentioning in the docs?

**Comment by robframpton** [comment]
Hey @mwilliams2014 

For the yeoman generator, there are these https://github.com/liferay/generator-liferay-theme/blob/dev/generators/app/index.js#L299.

I believe they just correspond to the prompts that you would otherwise follow when generating a new theme.

For the gulp tasks, there are the [custom paths](https://github.com/Robert-Frampton/liferay-theme-tasks/blob/master/lib/util.js#L27) to the theme dependencies (--unstyled-path, --styled-path, --css-common-path). Though I don't know how useful this would be for most developers as we made it specifically for calling from gradle tasks in liferay-portal.

**Comment by izaera** [comment]
@Robert-Frampton Not that I know. I just adapted some prompts to the new templating support, but the rest of the changes were just refactors, I think.

However, we will modify the tool in the future to implement the upgrade process, AFAIK. But still, I plan to change the less possible to avoid devs having to learn a new tool.

**Comment by mwilliams2014** [comment]
Thanks @Robert-Frampton and @izaera !

**Comment by mwilliams2014** [comment]
Hey @Robert-Frampton and @izaera was there any plans for including a feature in the theme generator that would upgrade your velocity templates to freemaker? I know we had briefly discussed that before, but I just wanted to check if it would be included in the upgrade process?

**Comment by izaera** [comment]
I don't think so. Migrating from vm to ftl is fairly easy (AFAIK) but having an option in the themes SDK would mean developing a complex tool because you need to implement a compiler. So, unless it is requested by PMs or something like that, I wouldn't recommend it.

My 2 cents

**Pull request closed without merging**